giskardpy/src/giskardpy/motion_statechart/tasks/cartesian_tasks.py

```diff
@@ -306,114 +309,171 @@ def on_start(self, context: ExecutionContext):
             self._fk_binding.bind(context.world)
 
 
-@dataclass
+@dataclass(eq=False, repr=False)
 class CartesianPositionVelocityLimit(Task):
-    root_link: Body = field(kw_only=True)
-    tip_link: Body = field(kw_only=True)
-    max_linear_velocity: float = 0.2
-    weight: float = DefaultWeights.WEIGHT_ABOVE_CA
+    """
+    Limit the Cartesian (translational) velocity of a tip link relative to a root link.
+
+    This goal enforces a strict cap on the linear speed of the frame defined by
+    the kinematic transform from `root_link` to `tip_link`. Enforcement is performed
+    by adding constraints to the optimizer and by providing an observation expression
+    that evaluates whether the current translational speed is within the limit.
+
+    .. warning::
+       Strict Cartesian velocity limits typically require many constraints to be
+       added to the optimization problem and can substantially slow down the
+       system. If runtime or responsiveness is important,
+       prefer a larger limit or a softer (lower-weight) constraint
```

Comment:
The last sentence is not true. I'm not talking about the slowdown due to slower motions. 

The optimization problem is getting more complex and therefore needs more time to solve. this may become a problem if Giskard is running at a very high frequency. A "normal" positional constraint, like we use for cartesian position only needs a single constraint in the underlying QP. But a velocity constraint requires as many constraints as our prediction horizon is large. so if the prediction horizon is 15, we need 15 constraints to properly enforce a velocity limit.

Reply:
yes